apps/console/src/features/applications/components/edit-application.tsx

```diff
@@ -333,6 +370,52 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
         setShowClientSecretHashDisclaimerModal(true);
     }, [ urlSearchParams.get(ApplicationManagementConstants.CLIENT_SECRET_HASH_ENABLED_URL_SEARCH_PARAM_KEY) ]);
 
+    /**
+     * Handles the defaultActiveIndex change.
+     */
+    const handleDefaultTabIndex = () => {
+        if ((application?.templateId === CustomApplicationTemplate.id
```

Comment:
Shall we just check if `application?.templateId` is defined at first. Then you can get rid of the optional chaining in the below if block.

```
if (!application?.templateId) {
    handleActiveTabIndexChange(defaultActiveIndex);

    return;
}
```

Reply:
Fixed